code/drasil-printers/lib/Language/Drasil/HTML/Helpers.hs

```diff
@@ -140,14 +140,19 @@ reflinkURI rf txt = text ("<a href=\"" ++ rf ++ "\">") <> txt <> text "</a>"
 
 -- | Helper for setting up figures.
 image :: Doc -> Doc -> MaxWidthPercent -> Doc
-image f c 100 = 
-  figure $ vcat [
-  img [("src", f), ("alt", c)],
-  figcaption c]
+image f c 100 =
+  figure $ vcat $ img [("src", f), ("alt", c)] : figCap c
 image f c wp =
-  figure $ vcat [
-  img [("src", f), ("alt", c), ("width", text $ show wp ++ "%")],
-  figcaption c]
+  figure $ vcat $ img [("src", f), ("alt", c), ("width", text $ show wp ++ "%")] : figCap c
+
+-- | Helper function for figcaption.
+figCap :: Doc -> [Doc]
```

Comment:
Seems like this choice is done 'too late' (i.e. it is already a `Doc`). I guess this could become a `Maybe Doc`. But basically we don't want to "dig into" a `Doc` to figure out if it is empty, but pass that information down explicitly.